src/client/mod.rs

```diff
@@ -196,6 +197,20 @@ impl GaloyClient {
         println!("going to execute:");
         batch.show();
 
+        if !force {
+            println!("do you want to confirm? y/n");
+
+            let mut input = String::new();
+
+            io::stdin()
+                .read_line(&mut input)
+                .expect("error: unable to read user input");
+
+            if !(input.trim() == "y" || input.trim() == "yes") {
+                bail!("didn't confirm batch payment. exiting")
+            }
```

Comment:
TBH I've never written any rust that had interaction like this so I don't know if there is a library or a standard convention on how to do this. Would have to go looking at other code to know if there is a better way.

Without doing any research I'd say this looks fine.

You could avoid the double call to `trim` but afaict it doesn't allocate so is negligible for this use-case.

Reply:
looking at https://lib.rs/command-line-interface, maybe dialoguer could be used https://github.com/mitsuhiko/dialoguer

Reply:
https://lib.rs/crates/dialoguer